File: backend/nem_data.py
FASES_GRADOS = {
    3: {"nombre": "Fase 3", "nivel": "Primaria", "grados": ["1°", "2°"], "descripcion": "Primaria baja"},
    4: {"nombre": "Fase 4", "nivel": "Primaria", "grados": ["3°", "4°"], "descripcion": "Primaria media"},
    5: {"nombre": "Fase 5", "nivel": "Primaria", "grados": ["5°", "6°"], "descripcion": "Primaria alta"},
    6: {"nombre": "Fase 6", "nivel": "Secundaria", "grados": ["1°", "2°", "3°"], "descripcion": "Secundaria"}
}

CAMPOS_FORMATIVOS = {
    "lenguajes": {
        "nombre": "Lenguajes",
        "disciplinas": ["Español", "Inglés", "Artes"],
        "metodologia": {
            "nombre": "Aprendizaje Basado en Proyectos Comunitarios (ABPC)",
            "fases": [
                {"nombre": "Planeación", "momentos": [
                    {"num": 1, "nombre": "Identificación", "descripcion": "Planteamiento del problema comunitario"},
                    {"num": 2, "nombre": "Recuperación", "descripcion": "Saberes previos: ¿qué sabemos?"},
                    {"num": 3, "nombre": "Planificación", "descripcion": "Actividades, tiempos, responsables, productos"}
                ]},
                {"nombre": "Acción", "momentos": [
                    {"num": 4, "nombre": "Acercamiento", "descripcion": "Exploración e investigación"},
                    {"num": 5, "nombre": "Comprensión y producción", "descripcion": "Desarrollo de contenidos y producciones"},
                    {"num": 6, "nombre": "Reconocimiento", "descripcion": "Valorar avances y ajustes"},
                    {"num": 7, "nombre": "Concreción", "descripcion": "Primera versión del producto + retroalimentación"}
                ]},
                {"nombre": "Intervención", "momentos": [
                    {"num": 8, "nombre": "Integración", "descripcion": "Exposición y retroalimentación final"},
                    {"num": 9, "nombre": "Difusión", "descripcion": "Compartir con la comunidad"},
                    {"num": 10, "nombre": "Consideraciones", "descripcion": "Instrumentos de seguimiento"},
                    {"num": 11, "nombre": "Avances", "descripcion": "Evaluación para proyectos futuros"}
                ]}
            ]
        }
    },
    "saberes": {
        "nombre": "Saberes y Pensamiento Científico",
        "disciplinas": ["Matemáticas", "Ciencias Naturales", "Tecnología"],
        "metodologia": {
            "nombre": "Aprendizaje Basado en Indagación (STEAM)",
            "fases": [
                {"num": 1, "nombre": "Introducción al tema", "descripcion": "Conocimientos previos e identificación de la problemática"},
                {"num": 2, "nombre": "Diseño de investigación", "descripcion": "Formulación de preguntas, planificación"},
                {"num": 3, "nombre": "Organizar y estructurar", "descripcion": "Sistematización de datos, análisis"},
                {"num": 4, "nombre": "Presentación de resultados", "descripcion": "Exposición de hallazgos, argumentación"},
                {"num": 5, "nombre": "Metacognición", "descripcion": "Reflexión individual, equipo y grupal"}
            ]
        }
    },
    "etica": {
        "nombre": "Ética, Naturaleza y Sociedades",
        "disciplinas": ["Historia", "Geografía", "Formación Cívica y Ética"],
        "metodologia": {
            "nombre": "Aprendizaje Basado en Problemas (ABP)",
            "fases": [
                {"num": 1, "nombre": "Presentemos", "descripcion": "Reflexión inicial, disparadores"},
                {"num": 2, "nombre": "Recolectemos", "descripcion": "Exploración de saberes previos"},
                {"num": 3, "nombre": "Formulemos el problema", "descripcion": "Definición clara del problema"},
                {"num": 4, "nombre": "Organicemos la experiencia", "descripcion": "Plan de acción: objetivos, recursos, tiempos"},
                {"num": 5, "nombre": "Vivamos la experiencia", "descripcion": "Ejecución e indagación"},
                {"num": 6, "nombre": "Resultados y análisis", "descripcion": "Revisión de hallazgos y divulgación"}
            ]
        }
    },
    "humano": {
        "nombre": "De lo Humano y lo Comunitario",
        "disciplinas": ["Educación Física", "Vida Saludable", "Educación Socioemocional"],
        "metodologia": {
            "nombre": "Aprendizaje Servicio (AS)",
            "fases": [
                {"num": 1, "nombre": "Punto de partida", "descripcion": "Identificación de problemática comunitaria"},
                {"num": 2, "nombre": "Lo que sé y lo que quiero saber", "descripcion": "Diagnóstico y saberes previos"},
                {"num": 3, "nombre": "Organicemos las actividades", "descripcion": "Planificación de acciones y recursos"},
                {"num": 4, "nombre": "Creatividad en marcha", "descripcion": "Ejecución y monitoreo"},
                {"num": 5, "nombre": "Compartimos y evaluamos", "descripcion": "Presentación y evaluación del impacto"}
            ]
        }
    }
}

EJES_ARTICULADORES = [
    "Inclusión", 
    "Pensamiento crítico", 
    "Interculturalidad crítica", 
    "Igualdad de género", 
    "Vida saludable", 
    "Apropiación de las culturas a través de la lectura y la escritura", 
    "Artes y experiencias estéticas"
]

CONTENIDOS_PDA = {
    3: {  # Fase 3 (1° y 2° Primaria)
        "lenguajes": [
            {
                "contenido": "Uso de narraciones de la tradición oral de la comunidad",
                "pdas": [
                    "Narra de forma oral experiencias personales y comunitarias",
                    "Identifica la estructura de textos narrativos sencillos",
                    "Escribe textos breves a partir de sus vivencias"
                ]
            },
            {
                "contenido": "Exploración de la diversidad de lenguas en su comunidad",
                "pdas": [
                    "Reconoce que existen diferentes formas de nombrar las cosas",
                    "Participa en juegos de palabras en diversas lenguas",
                    "Valora la riqueza lingüística de su entorno"
                ]
            },
            {
                "contenido": "Expresión artística y apreciación de manifestaciones estéticas",
                "pdas": [
                    "Explora materiales y técnicas de artes visuales",
                    "Expresa emociones mediante lenguajes artísticos",
                    "Aprecia manifestaciones artísticas de su comunidad"
                ]
            }
        ],
        "saberes": [
            {
                "contenido": "Estudio de los números y suma y resta",
                "pdas": [
                    "Cuenta objetos y elementos de su entorno",
                    "Resuelve problemas de suma y resta con objetos reales",
                    "Identifica el valor posicional en números de dos cifras"
                ]
            },
            {
                "contenido": "Conocimiento del cuerpo humano y sus cuidados",
                "pdas": [
                    "Identifica las partes externas del cuerpo humano",
                    "Comprende la importancia de la higiene personal",
                    "Reconoce medidas de prevención de accidentes"
                ]
            },
            {
                "contenido": "Exploración de fenómenos naturales",
                "pdas": [
                    "Observa y describe cambios en el clima",
                    "Identifica características de plantas y animales de su entorno",
                    "Formula preguntas sobre cómo funcionan las cosas"
                ]
            }
        ],
        "etica": [
            {
                "contenido": "La familia como espacio de protección, cuidado y afecto",
                "pdas": [
                    "Reconoce la importancia de la familia",
                    "Identifica responsabilidades en el hogar",
                    "Expresa cómo se siente en su familia"
                ]
            },
            {
                "contenido": "Historia de la comunidad y costumbres",
                "pdas": [
                    "Narra historias de su comunidad",
                    "Identifica costumbres y tradiciones locales",
                    "Compara objetos del pasado y del presente"
                ]
            },
            {
                "contenido": "Cuidado de los ecosistemas",
                "pdas": [
                    "Identifica problemas ambientales en su entorno",
                    "Participa en acciones para cuidar el medio ambiente",
                    "Reconoce a los seres vivos y sus necesidades"
                ]
            }
        ],
        "humano": [
            {
                "contenido": "Desarrollo de habilidades motrices",
                "pdas": [
                    "Explora diferentes formas de movimiento",
                    "Participa en juegos colectivos",
                    "Mantiene el equilibrio en situaciones lúdicas"
                ]
            },
            {
                "contenido": "Reconocimiento y gestión de emociones",
                "pdas": [
                    "Identifica emociones básicas en sí mismo y en otros",
                    "Expresa sus emociones de manera asertiva",
                    "Muestra empatía ante las emociones de sus compañeros"
                ]
            },
            {
                "contenido": "Hábitos de vida saludable",
                "pdas": [
                    "Reconoce alimentos saludables y comida chatarra",
                    "Practica el lavado de manos correctamente",
                    "Comprende la importancia del descanso y la actividad física"
                ]
            }
        ]
    },
    # Para brevedad, el resto de fases (4, 5, 6) pueden ser pobladas con datos similares
    4: { # Fase 4 (3° y 4° Primaria)
        "lenguajes": [
            {
                "contenido": "Análisis de textos informativos",
                "pdas": ["Identifica la idea principal de un texto informativo", "Elabora resúmenes breves", "Usa fuentes de información para investigar un tema"]
            }
        ],
        "saberes": [
            {
                "contenido": "Multiplicación y división",
                "pdas": ["Resuelve problemas de multiplicación", "Comprende la relación entre multiplicación y división", "Utiliza algoritmos convencionales"]
            }
        ],
        "etica": [
            {
                "contenido": "Historia de la entidad federativa",
                "pdas": ["Identifica los primeros habitantes de su entidad", "Reconoce cambios en el paisaje a lo largo del tiempo", "Valora el patrimonio cultural"]
            }
        ],
        "humano": [
            {
                "contenido": "Resolución de conflictos",
                "pdas": ["Identifica causas de los conflictos", "Propone soluciones pacíficas", "Practica el diálogo respetuoso"]
            }
        ]
    },
    5: { # Fase 5 (5° y 6° Primaria)
        "lenguajes": [
             {
                "contenido": "Textos argumentativos y artículos de opinión",
                "pdas": ["Estructura un texto argumentativo", "Expresa una opinión fundamentada", "Diferencia entre hechos y opiniones"]
             }
        ],
        "saberes": [
             {
                "contenido": "Fracciones y decimales",
                "pdas": ["Resuelve problemas con fracciones", "Opera con números decimales", "Aplica porcentajes en situaciones reales"]
             }
        ],
        "etica": [
             {
                "contenido": "México y el mundo",
                "pdas": ["Ubica a México en el contexto mundial", "Analiza problemas globales", "Reconoce la diversidad cultural del mundo"]
             }
        ],
        "humano": [
             {
                "contenido": "Proyecto de vida",
                "pdas": ["Establece metas a corto y mediano plazo", "Identifica fortalezas y áreas de oportunidad", "Toma decisiones responsables"]
             }
        ]
    },
    6: { # Fase 6 (Secundaria)
        "lenguajes": [
             {
                "contenido": "La diversidad de lenguas y su uso en la comunicación familiar, escolar y comunitaria",
                "pdas": ["Reconoce la riqueza lingüística de México", "Analiza el uso de la lengua en distintos contextos", "Valora la importancia de preservar las lenguas indígenas"]
             }
        ],
        "saberes": [
             {
                "contenido": "Ecuaciones lineales y cuadráticas",
                "pdas": ["Resuelve ecuaciones de primer grado", "Plantea problemas usando álgebra", "Grafica funciones lineales"]
             }
        ],
        "etica": [
             {
                "contenido": "Las revoluciones modernas y sus tendencias",
                "pdas": ["Analiza las causas de la Revolución Francesa", "Comprende el impacto de la Revolución Industrial", "Relaciona eventos históricos con el presente"]
             }
        ],
        "humano": [
             {
                "contenido": "Sexualidad y equidad de género",
                "pdas": ["Reconoce los derechos sexuales y reproductivos", "Promueve la equidad de género", "Toma decisiones informadas sobre su cuerpo"]
             }
        ]
    }
}

INSTRUMENTOS_EVALUACION = [
    "Rúbrica", 
    "Lista de cotejo", 
    "Portafolio de evidencias", 
    "Diario de clase", 
    "Registro anecdótico", 
    "Escala de actitudes", 
    "Guía de observación"
]

TIPOS_EVALUACION = ["Diagnóstica", "Formativa", "Sumativa"]

ESCENARIOS = ["Aula", "Escuela", "Comunidad"]

File: frontend/js/nem-data.js
window.NEM = {
    FASES: {
        3: { nombre: "Fase 3", grados: ["1°", "2°"], nivel: "Primaria" },
        4: { nombre: "Fase 4", grados: ["3°", "4°"], nivel: "Primaria" },
        5: { nombre: "Fase 5", grados: ["5°", "6°"], nivel: "Primaria" },
        6: { nombre: "Fase 6", grados: ["1°", "2°", "3°"], nivel: "Secundaria" }
    },
    
    CAMPOS: {
        lenguajes: {
            nombre: "Lenguajes",
            metodologia: {
                nombre: "Aprendizaje Basado en Proyectos Comunitarios (ABPC)",
                fases: [
                    { nombre: "Fase 1: Planeación", momentos: ["1. Identificación", "2. Recuperación", "3. Planificación"] },
                    { nombre: "Fase 2: Acción", momentos: ["4. Acercamiento", "5. Comprensión y producción", "6. Reconocimiento", "7. Concreción"] },
                    { nombre: "Fase 3: Intervención", momentos: ["8. Integración", "9. Difusión", "10. Consideraciones", "11. Avances"] }
                ]
            }
        },
        saberes: {
            nombre: "Saberes y Pensamiento Científico",
            metodologia: {
                nombre: "Aprendizaje Basado en Indagación (STEAM)",
                fases: [
                    { nombre: "Fase 1", momentos: ["Introducción al tema / Conocimientos previos"] },
                    { nombre: "Fase 2", momentos: ["Diseño de investigación / Formulación de preguntas"] },
                    { nombre: "Fase 3", momentos: ["Organizar y estructurar las respuestas"] },
                    { nombre: "Fase 4", momentos: ["Presentación de resultados"] },
                    { nombre: "Fase 5", momentos: ["Metacognición / Reflexión"] }
                ]
            }
        },
        etica: {
            nombre: "Ética, Naturaleza y Sociedades",
            metodologia: {
                nombre: "Aprendizaje Basado en Problemas (ABP)",
                fases: [
                    { nombre: "Momento 1", momentos: ["Presentemos"] },
                    { nombre: "Momento 2", momentos: ["Recolectemos"] },
                    { nombre: "Momento 3", momentos: ["Formulemos el problema"] },
                    { nombre: "Momento 4", momentos: ["Organicemos la experiencia"] },
                    { nombre: "Momento 5", momentos: ["Vivamos la experiencia"] },
                    { nombre: "Momento 6", momentos: ["Resultados y análisis"] }
                ]
            }
        },
        humano: {
            nombre: "De lo Humano y lo Comunitario",
            metodologia: {
                nombre: "Aprendizaje Servicio (AS)",
                fases: [
                    { nombre: "Etapa 1", momentos: ["Punto de partida"] },
                    { nombre: "Etapa 2", momentos: ["Lo que sé y lo que quiero saber"] },
                    { nombre: "Etapa 3", momentos: ["Organicemos las actividades"] },
                    { nombre: "Etapa 4", momentos: ["Creatividad en marcha"] },
                    { nombre: "Etapa 5", momentos: ["Compartimos y evaluamos lo aprendido"] }
                ]
            }
        }
    },
    
    EJES: [
        "Inclusión", 
        "Pensamiento crítico", 
        "Interculturalidad crítica", 
        "Igualdad de género", 
        "Vida saludable", 
        "Apropiación de las culturas a través de la lectura y la escritura", 
        "Artes y experiencias estéticas"
    ],
    
    CONTENIDOS: {
        3: {
            lenguajes: [
                { id: "L3_1", texto: "Uso de narraciones de la tradición oral de la comunidad", pdas: ["Narra de forma oral experiencias personales", "Identifica estructura de textos narrativos", "Escribe textos breves"] },
                { id: "L3_2", texto: "Exploración de la diversidad de lenguas", pdas: ["Reconoce diferentes formas de nombrar cosas", "Participa en juegos de palabras", "Valora la riqueza lingüística"] }
            ],
            saberes: [
                { id: "S3_1", texto: "Estudio de los números y suma y resta", pdas: ["Cuenta objetos del entorno", "Resuelve problemas de suma y resta", "Identifica valor posicional"] },
                { id: "S3_2", texto: "Conocimiento del cuerpo humano y cuidados", pdas: ["Identifica partes externas del cuerpo", "Comprende higiene personal", "Reconoce medidas de prevención"] }
            ],
            etica: [
                { id: "E3_1", texto: "La familia como espacio de protección", pdas: ["Reconoce importancia de la familia", "Identifica responsabilidades", "Expresa sentimientos familiares"] },
                { id: "E3_2", texto: "Cuidado de los ecosistemas", pdas: ["Identifica problemas ambientales", "Participa en acciones ecológicas", "Reconoce seres vivos"] }
            ],
            humano: [
                { id: "H3_1", texto: "Desarrollo de habilidades motrices", pdas: ["Explora formas de movimiento", "Participa en juegos colectivos", "Mantiene equilibrio"] },
                { id: "H3_2", texto: "Reconocimiento y gestión de emociones", pdas: ["Identifica emociones básicas", "Expresa emociones asertivamente", "Muestra empatía"] }
            ]
        },
        4: {
            lenguajes: [
                { id: "L4_1", texto: "Análisis de textos informativos", pdas: ["Identifica idea principal", "Elabora resúmenes", "Usa fuentes de información"] }
            ],
            saberes: [
                { id: "S4_1", texto: "Multiplicación y división", pdas: ["Resuelve problemas de multiplicación", "Comprende relación multiplicación-división", "Usa algoritmos convencionales"] }
            ],
            etica: [
                { id: "E4_1", texto: "Historia de la entidad federativa", pdas: ["Identifica primeros habitantes", "Reconoce cambios en el paisaje", "Valora patrimonio cultural"] }
            ],
            humano: [
                { id: "H4_1", texto: "Resolución de conflictos", pdas: ["Identifica causas", "Propone soluciones pacíficas", "Practica diálogo"] }
            ]
        },
        5: {
            lenguajes: [
                { id: "L5_1", texto: "Textos argumentativos y artículos de opinión", pdas: ["Estructura texto argumentativo", "Expresa opinión fundamentada", "Diferencia hechos de opiniones"] }
            ],
            saberes: [
                { id: "S5_1", texto: "Fracciones y decimales", pdas: ["Resuelve problemas con fracciones", "Opera con decimales", "Aplica porcentajes"] }
            ],
            etica: [
                { id: "E5_1", texto: "México y el mundo", pdas: ["Ubica a México en contexto mundial", "Analiza problemas globales", "Reconoce diversidad cultural"] }
            ],
            humano: [
                { id: "H5_1", texto: "Proyecto de vida", pdas: ["Establece metas", "Identifica fortalezas", "Toma decisiones"] }
            ]
        },
        6: {
            lenguajes: [
                { id: "L6_1", texto: "La diversidad de lenguas", pdas: ["Reconoce riqueza lingüística", "Analiza uso de la lengua", "Valora lenguas indígenas"] }
            ],
            saberes: [
                { id: "S6_1", texto: "Ecuaciones lineales y cuadráticas", pdas: ["Resuelve ecuaciones 1er grado", "Plantea problemas algebraicos", "Grafica funciones"] }
            ],
            etica: [
                { id: "E6_1", texto: "Revoluciones modernas", pdas: ["Analiza Revolución Francesa", "Comprende Revolución Industrial", "Relaciona eventos históricos"] }
            ],
            humano: [
                { id: "H6_1", texto: "Sexualidad y equidad de género", pdas: ["Reconoce derechos", "Promueve equidad", "Toma decisiones informadas"] }
            ]
        }
    },
    
    INSTRUMENTOS: [
        "Rúbrica", "Lista de cotejo", "Portafolio de evidencias", 
        "Diario de clase", "Registro anecdótico", "Escala de actitudes", "Guía de observación"
    ],
    
    EVALUACION_TIPOS: ["Diagnóstica", "Formativa", "Sumativa"],
    ESCENARIOS: ["Aula", "Escuela", "Comunidad"]
};
